paper/src/main/java/com/denizenscript/denizen/paper/events/PlayerClientOptionsChangeScriptEvent.java

```diff
@@ -0,0 +1,100 @@
+import org.bukkit.event.Listener;
+
+public class PlayerClientOptionsChangeScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // player client options change
+    //
+    // @Plugin Paper
+    //
+    // @Group Paper
+    //
+    // @Player Always.
+    //
+    // @Triggers when a player changes their client options.
+    //
+    // @Context
+    // <context.server_listings_enabled> returns an ElementTag(Boolean) of whether the player has server listings enabled. Available only on MC 1.19+.
+    // <context.chat_visibility> returns the player's chat visibility.
+    // <context.locale> returns the player's locale.
+    // <context.main_hand> returns an ElementTag(Boolean) of whether the player's main hand has changed.
+    // <context.skin_parts> returns a MapTag of whether the player's skin parts are enabled or not.
+    // <context.view_distance> returns the player's view distance.
+    // <context.server_listings_changed> returns whether the player's server listings have changed. Available only on MC 1.19+.
+    // <context.chat_colors> returns whether the player has chat colors enabled.
+    // <context.chat_colors_changed> returns whether the player's chat colors have changed.
+    // <context.chat_visibility_changed> returns the player's chat visibility. Available only on MC 1.19+.
+    // <context.locale_changed> returns whether the player's locale has changed.
+    // <context.main_hand_changed> returns whether the player's main hand has changed.
+    // <context.skin_parts_changed> returns whether the player's skin parts have changed.
+    // <context.text_filtering_changed> returns whether the player's text filtering has changed. Available only on MC 1.19+.
+    // <context.text_filtering_enabled> returns whether the player has text filtering enabled. Available only on MC 1.19+.
+    // <context.view_distance_changed> returns whether the player's view distance has changed.
+    //
+    // @Player Always.
+    //
+    // -->
+
+    public PlayerClientOptionsChangeScriptEvent() {
+        registerCouldMatcher("player client options change");
+    }
+
+    public PlayerClientOptionsChangeEvent event;
+
+    @Override
+    public ScriptEntryData getScriptEntryData() {
+        return new BukkitScriptEntryData(event.getPlayer());
+    }
+
+    @Override
+    public ObjectTag getContext(String name) {
+        return switch (name) {
+            case "server_listings_enabled" -> NMSHandler.getVersion().isAtLeast(NMSVersion.v1_19) ? new ElementTag(event.allowsServerListings()) : null;
+            case "chat_visibility" -> new ElementTag(event.getChatVisibility());
+            case "locale" -> new ElementTag(event.getLocale());
+            case "main_hand" -> new ElementTag(event.getMainHand());
+            case "skin_parts" -> {
+                MapTag map = new MapTag();
+                map.putObject("cape", new ElementTag(event.getSkinParts().hasCapeEnabled()));

```

Comment:
grab the `getSkinParts()` return and put it in a variable, so you're not calling the same `getSkinParts` method 7 times in a row